benchmarks/src/main/java/com/slack/kaldb/IndexTraceLogBenchmark.java

```diff
@@ -0,0 +1,185 @@
+  //  static {
+  //    String traceLogFile = System.getProperty("jmh.trace.log.file", "trace_logs.txt");
+  //    try {
+  //      createMurronTraceLogFile(traceLogFile);
+  //    } catch (IOException e) {
+  //      System.out.println(e);
+  //    }
+  //  }
+  //
+  //  // We wanted to write out this intermediary file so that the benchmarks are more accurate
+  //  // However I haven't figured out out a good way to write it to a file such that we can read it
+  // ( ser-de )
+  //  // So for now i am commenting out this approach
+  //  private static void createMurronTraceLogFile(String traceLogFile) throws IOException {
+  //
+  //    try (BufferedReader reader =
+  //            Files.newBufferedReader(Path.of(traceLogFile), StandardCharsets.UTF_8);
+  //        BufferedWriter writer =
+  //            Files.newBufferedWriter(
+  //                Path.of("trace_logs_murron.txt"),
+  //                StandardCharsets.UTF_8,
+  //                StandardOpenOption.CREATE_NEW)) {
+  //      String message;
+  //      try {
+  //        while ((message = reader.readLine()) != null) {
+  //          Trace.ListOfSpans.Builder builder = Trace.ListOfSpans.newBuilder();
+  //          JsonFormat.parser().merge(message, builder);
+  //          Trace.ListOfSpans spans = builder.build();
+  //
+  //          writer.write(String.valueOf(spans.getSpans(0).getStartTimestampMicros()));
+  //          writer.write(" ");
+  //          writer.write(spans.toByteString().toString(StandardCharsets.UTF_8));
+  //          writer.newLine();
+  //        }
+  //
+  //        // So that we don't run out of disk spaces
+  //        Files.delete(Path.of(traceLogFile));
+  //      } catch (IOException e) {
+  //        skipCount++;
+  //      }
+  //      System.out.println("Unable to parse " + skipCount + " lines");
+  //    }
+  //  }
+
+  @Setup(Level.Iteration)
+  public void createIndexer() throws Exception {
+
+    //        reader = Files.newBufferedReader(Path.of("trace_logs.txt"),
+    // StandardCharsets.UTF_8);
+
+    String traceLogFile = System.getProperty("jmh.trace.log.file", "trace_logs.txt");
+    reader = Files.newBufferedReader(Path.of(traceLogFile));
+
+    random = new Random();
+    registry = new SimpleMeterRegistry();
+    tempDirectory =
+        Files.createDirectories(
+            Paths.get("jmh-output", String.valueOf(random.nextInt(Integer.MAX_VALUE))));
+    logStore =
+        LuceneIndexStoreImpl.makeLogStore(
+            tempDirectory.toFile(), commitInterval, refreshInterval, registry);
+
+    skipCount = 0;
+    indexCount = 0;
+  }
+
+  @TearDown(Level.Iteration)
+  public void tearDown() throws IOException {
+    Directory directory = logStore.getIndexWriter().getDirectory();
+    String[] segmentFiles = directory.listAll();
+    long indexedBytes = 0;
+    for (String segmentFile : segmentFiles) {
+      indexedBytes += directory.fileLength(segmentFile);
+    }
+    System.out.println(
```

Comment:
Are we using the SLF4J logger for the tests? If so we should probably use it here. If not, why do we have a [benchmark specific SLF4J config](https://github.com/slackhq/kaldb/blob/master/benchmarks/log4j2.xml)?

Reply:
very good point! haha I had forgotten

```
[INFO ] 2021-09-22 13:28:46.221 [com.slack.kaldb.IndexTraceLogBenchmark.measureTraceLogIndexingSlingshotMode-jmh-worker-1]  IndexTraceLogBenchmark - Indexed = 100 Skipped = 0 Index size = 47 KB

```

using the logger and output gets printed to console